common/src/main/java/com/hedera/block/common/constants/StringConstants.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.common.constants;
+
+import static com.hedera.block.common.constants.ErrorMessageConstants.CREATING_INSTANCES_NOT_SUPPORTED;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/** A class that hold common {@link String} literals used across projects. */
+public final class StringConstants {
+    // FILES
+    @NonNull public static final String APPLICATION_PROPERTIES = "app.properties";
+    @NonNull public static final String LOGGING_PROPERTIES = "logging.properties";
+
+    // FILE EXTENSIONS
+    @NonNull public static final String BLOCK_FILE_EXTENSION = ".blk";
+    @NonNull public static final String GZ_FILE_EXTENSION = ".gz";
+
+    // PROJECT RELATED
+    @NonNull public static final String SERVICE_NAME = "BlockStreamService";
+    @NonNull public static final String CLIENT_STREAMING_METHOD_NAME = "publishBlockStream";
+    @NonNull public static final String SERVER_STREAMING_METHOD_NAME = "subscribeBlockStream";
+    @NonNull public static final String SINGLE_BLOCK_METHOD_NAME = "singleBlock";
+
+    private StringConstants() {
+        throw new UnsupportedOperationException(CREATING_INSTANCES_NOT_SUPPORTED);
+    }
```

Comment:
could be simpler to do 
```suggestion
    private StringConstants() { }
```
with a comment that this is a utility class, so that constructor is unreachable, no instances are supported.

I think that throwing an error does not apply, since is literally impossible to reach that code 😅

Reply:
As a best practice, an error must be thrown, because we could without problem create an instance of this class using reflection, no matter that the constructor is private. When we have an exception thrown, this is no longer the case, since it will result in an invocation exception. I know this is a wild far fetch, but mistakes could happen, especially if some logic that creates classes at runtime is introduced directly by us or some library. We never know what would happen in the future, but we for sure want to catch if instance of classes that are not supposed to be created actually get created. Also, this will never (hopefully) be called ever, so in no way does it hinder any performance whatsoever. I would disagree here with your suggestion.

Reply:
how are we going to UT this?

We are missing tests on this PR.

Reply:
You mean unit tests for the whole commit in general? If so, I was thinking to do it in the next commits that are related to this based on the epic #254 since the logic here is not used yet. My goal is to ship this ASAP. If that seems as a blocker to you, I could add unit tests now. Let me know.

Reply:
The exception throw is removed.